### PR TITLE
RDCC-1567 - HMCTS checkstyle Plugin in profile sync

### PR DESCRIPTION
**Pull request by pksaxena01**
### JIRA link (if applicable) ###
RDCC-1567

### Change description ###
This pull request add the HMCTS checkstyle plugin which will help to reduce the warnings

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -347,11 +340,8 @@ dependencies {
 
 // https://jeremylong.github.io/DependencyCheck/dependency-check-gradle/configuration.html
 dependencyCheck {
-	// Specifies if the build should be failed if a CVSS score above a specified level is identified.
-	// range of 0-10 fails the build, anything greater and it doesn't fail the build
-	//FIXME
-	failBuildOnCVSS = System.getProperty('dependencyCheck.failBuild') == 'true' ? 0 : 11
```
Comment (AdilOoze) [inline build.gradle:353]: any reason we've removed the build failure if CVSS score is below 11?

File: src/main/java/uk/gov/hmcts/reform/profilesync/util/JsonFeignResponseUtil.java [hunk 2]
```diff
@@ -22,15 +22,17 @@
 @SuppressWarnings("unchecked")
 @Slf4j
 public class JsonFeignResponseUtil {
-    private static final ObjectMapper json = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    private static final ObjectMapper json = new ObjectMapper().configure(
+            DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```
Comment (AdilOoze) [inline src/main/java/uk/gov/hmcts/reform/profilesync/util/JsonFeignResponseUtil.java:26]: _(Non-essential)_ Could you move `.configure(` onto the next line also so it reads better?

File: src/main/java/uk/gov/hmcts/reform/profilesync/client/UserProfileClient.java [hunk 3]
```diff
@@ -20,7 +20,8 @@
 @FeignClient(name = "userProfileClient", url = "${userprofile.api.url}")
 public interface UserProfileClient {
 
-    @PutMapping(value = "/v1/userprofile/{userId}", consumes = {APPLICATION_JSON_VALUE}, produces = {APPLICATION_JSON_VALUE})
+    @PutMapping(value = "/v1/userprofile/{userId}", consumes = {APPLICATION_JSON_VALUE}, produces =
+            {APPLICATION_JSON_VALUE})
     @Headers({"authorization: {authorization}", "serviceauthorization: {serviceauthorization}"})
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/profilesync/client/UserProfileClient.java:25]: can't see any difference..can we plz revert this in no difference. as it adds unchanged file in change list.

File: src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileSyncServiceImpl.java [hunk 4]
```diff
@@ -131,7 +133,8 @@ public String getS2sToken() {
     public void updateUserProfileFeed(String searchQuery) throws UserProfileSyncException {
         log.info("Inside updateUserProfileFeed");
         String bearerToken = BEARER + getBearerToken();
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileSyncServiceImpl.java:135]: Please avoid + use  concat.

File: src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileUpdateServiceImpl.java [hunk 5]
```diff
@@ -51,7 +53,8 @@ public void updateUserProfile(String searchQuery, String bearerToken, String s2s
                         .email(user.getEmail())
                         .firstName(user.getForename())
                         .lastName(user.getSurname())
-                        .idamStatus(idamStatusResolver().get(status) != null ? idamStatusResolver().get(status).name() : IdamStatus.SUSPENDED.name())
+                        .idamStatus(idamStatusResolver().get(status) != null ? idamStatusResolver().get(status).name()
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileUpdateServiceImpl.java:56]: StringUtils.empty we can use here.

File: src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileUpdateServiceImpl.java [hunk 6]
```diff
@@ -77,7 +80,8 @@ private void syncUser(String bearerToken, String s2sToken,
         log.info("Body response::" + response.body().toString());
         if (response.status() > 300) {
 
-            log.error("Exception occurred while updating the user profile: Status - {}" + userId + ":" + updatedUserProfile.getIdamStatus());
+            log.error("Exception occurred while updating the user profile: Status - {}" + userId + ":"
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileUpdateServiceImpl.java:83]: Please remove + I think with {} no need +. Use below

 log.error("Exception occurred while updating the user profile: Status - {}" ,userId);

File: build.gradle (hunk 1 above)
Comment (pksaxena01) [reply build.gradle:353]: This has been now handled by HMCTS plugin.

File: src/main/java/uk/gov/hmcts/reform/profilesync/util/JsonFeignResponseUtil.java (hunk 2 above)
Comment (pksaxena01) [reply src/main/java/uk/gov/hmcts/reform/profilesync/util/JsonFeignResponseUtil.java:26]: Done.

File: src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileSyncServiceImpl.java (hunk 4 above)
Comment (pksaxena01) [reply src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileSyncServiceImpl.java:135]: Thanks Abhijit. I have now used concat()for prod code.

File: src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileUpdateServiceImpl.java (hunk 6 above)
Comment (pksaxena01) [reply src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileUpdateServiceImpl.java:83]: Done

**Pull request merged**